File: RuntimeData/xnaruntime/RuntimeXNA/RuntimeXNA/Actions/ACT_SETINPUTKEY.cs
// -----------------------------------------------------------------------------
//
// SET INPUT KEY
//
// -----------------------------------------------------------------------------
using System;
using RuntimeXNA.RunLoop;
using RuntimeXNA.Params;
namespace RuntimeXNA.Actions
{
	
	public class ACT_SETINPUTKEY:CAct
	{
		public override void  execute(CRun rhPtr)
		{
			int touche = rhPtr.get_EventExpressionInt((CParamExpression) evtParams[0]); //; Numero de la touche
			if (touche >= 8)
			// MAX_KEY
				return ;
			int joueur = evtOi;
			if (joueur >= 4)
			// MAX_PLAYER 
				return ;
			rhPtr.rhApp.pcCtrlKeys[joueur*4+touche] = ((PARAM_KEY) evtParams[1]).key; // Nouvelle touche
		}
	}
}

File: RuntimeData/runtime/app/src/main/java/Actions/ACT_SETINPUTKEY.java
/* Copyright (c) 1996-2013 Clickteam
 *
 * This source code is part of the Android exporter for Clickteam Multimedia Fusion 2.
 * 
 * Permission is hereby granted to any person obtaining a legal copy 
 * of Clickteam Multimedia Fusion 2 to use or modify this source code for 
 * debugging, optimizing, or customizing applications created with 
 * Clickteam Multimedia Fusion 2.  Any other use of this source code is prohibited.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 * IN THE SOFTWARE.
 */
// -----------------------------------------------------------------------------
//
// SET INPUT KEY
//
// -----------------------------------------------------------------------------
package Actions;

import Params.CParamExpression;
import Params.PARAM_KEY;
import RunLoop.CRun;

public class ACT_SETINPUTKEY extends CAct
{
    public void execute(CRun rhPtr)
    {
	int touche=rhPtr.get_EventExpressionInt((CParamExpression)evtParams[0]);	//; Numero de la touche
	if (touche>=8)		// MAX_KEY
	    return;
	int joueur=evtOi;
	if ( joueur >= 4 )	// MAX_PLAYER 
	    return;
	rhPtr.rhApp.pcCtrlKeys[joueur][touche]=((PARAM_KEY)evtParams[1]).key;	// Nouvelle touche
    }
}
